apis/r/R/SparseReadIter.R

```diff
@@ -0,0 +1,80 @@
+#' SparseReadIter
+#'
+#' @description
+#' \code{SparseReadIter} is a class that allows for iteration over 
+#'  a reads on \link{SOMASparseNDArray}.
+#' Iteration chunks are retrieved as 0-based Views \link{matrixZeroBasedView} of Matrix::\link[Matrix]{SparseMatrix}.
+#' @export
+
+SparseReadIter <- R6::R6Class(
+  classname = "SparseReadIter",
+  inherit = ReadIter,
+
+  public = list(
+                
+    #' @description Create (lifecycle: experimental)
+    #' @param uri Character value with URI path to a SOMADataFrame or SOMASparseNDArray
+    #' @param config character vector containing TileDB config.
+    #' @param colnames Optional vector of character value with the name of the columns to retrieve
+    #' @param qc Optional external Pointer object to TileDB Query Condition, defaults to \sQuote{NULL} i.e.
+    #' no query condition
+    #' @param dim_points Optional named list with vector of data points to select on the given
+    #' dimension(s). Each dimension can be one entry in the list.
+    #' @param loglevel Character value with the desired logging level, defaults to \sQuote{auto}
+    #' @param repr Optional one-character code for sparse matrix representation type
+    #' which lets prior setting prevail, any other value is set as new logging level.
+    initialize = function(uri, 
+                          config, 
+                          colnames = NULL, 
+                          qc = NULL, 
+                          dim_points = NULL, 
+                          loglevel = "auto", 
+                          repr = c("C", "T", "R")) {
+        
+        # Initiate super class
+          super$initialize (uri = uri, config = config, colnames = colnames, qc = qc,
+                            dim_points = dim_points, loglevel = loglevel)
+    
+          private$repr <- repr
+          
+          # Get max soma dims for indeces via tiledb
+          tiledb_array <- tiledb::tiledb_array(uri)
+          tiledb::tiledb_array_open(tiledb_array, type = "READ")
+          max_soma_dim_0 <- as.integer(max(tiledb::tiledb_array_get_non_empty_domain_from_index(tiledb_array, 1)))
```

Comment:
assert this is a 32-byte AND 

get  max_soma_dom_0 - min_soma_dim_0 as the worst-case cardinality

Reply:
obsolete comment